snapcraft/plugins/go.py

```diff
@@ -191,4 +197,20 @@ def _build_environment(self):
         env['CGO_LDFLAGS'] = '{} {} {}'.format(
             env.get('CGO_LDFLAGS', ''), flags, env.get('LDFLAGS', ''))
 
+        if self.project.is_cross_compiling:
+            env['CC'] = '{}-gcc'.format(self.project.arch_triplet)
+            env['CXX'] = '{}-g++'.format(self.project.arch_triplet)
+            env['CGO_ENABLED'] = '1'
+            # See https://golang.org/doc/install/source#environment
+            go_archs = {
+                'armhf': 'arm',
+                'i386': '386',
+                'ppc64el': 'ppc64le',
+            }
+            env['GOARCH'] = go_archs.get(self.project.deb_arch,
+                                         self.project.deb_arch)
+            env['GOARM'] = '7'
```

Comment:
I would add an `if self.project.debarch == 'armhf'` here, `GOARM` is probably 8 for arm64 anyways